salt/minion.py

```diff
@@ -2591,7 +2591,8 @@ def _post_master_init(self, master):
 
         proxy_init_fn = self.proxy[fq_proxyname+'.init']
         proxy_init_fn(self.opts)
-        self.opts['grains'] = salt.loader.grains(self.opts)
+        self.opts['grains'] = salt.loader.grains(self.opts, force_refresh=True)
+        self.functions['saltutil.sync_grains'](saltenv='base')
```

Comment:
Proxies have a chicken and egg problem.  We can't retrieve their grains until the proxymodule is setup and talking to the remote device.  So I moved the grains loading here and I needed to force a grains sync because the master wouldn't target via grains until we did.